colour/quality/cfi2017.py

```diff
@@ -0,0 +1,405 @@
+        Spectral distribution of the reference illuminant.
+    R_f : numeric
+        *CIE 2017 Colour Fidelity Index* (CFI) :math:`R_f`.
+    R_s : array_like
+        Individual *colour fidelity indexes* data for each sample.
+    CCT : numeric
+        Correlated colour temperature :math:`T_{cp}`.
+    D_uv : numeric
+        Distance from the Planckian locus :math:`\\Delta_{uv}`.
+    colorimetry_data : tuple
+        Colorimetry data for the test and reference computations.
+    delta_E_s : ndarray, (16,)
+        Colour shifts of samples.
+    """
+
+
+def colour_fidelity_index_CIE2017(sd_test, additional_data=False):
+    """
+    Returns the *CIE 2017 Colour Fidelity Index* (CFI) :math:`R_f` of given
+    spectral distribution.
+
+    Parameters
+    ----------
+    sd_test : SpectralDistribution
+        Test spectral distribution.
+    additional_data : bool, optional
+        Whether to output additional data.
+
+    Returns
+    -------
+    numeric or ColourRendering_Specification_CIE2017
+        *CIE 2017 Colour Fidelity Index* (CFI) :math:`R_f`.
+
+    References
+    ----------
+    :cite:`CIETC1-902017`
+
+    Examples
+    --------
+    >>> from colour.colorimetry import SDS_ILLUMINANTS
+    >>> sd = SDS_ILLUMINANTS['FL2']
+    >>> colour_fidelity_index_CIE2017(sd)  # doctest: +ELLIPSIS
+    70.1208254...
+    """
+
+    if sd_test.shape.start > 380 or sd_test.shape.end < 780:
+        usage_warning('Test spectral distribution shape does not span the'
+                      'recommended 380-780nm range, missing values will be'
+                      'filled with zeros!')
+
+        # NOTE: "CIE 2017 Colour Fidelity Index" standard recommends filling
+        # missing values with zeros.
+        sd_test = sd_test.copy()
+        sd_test.extrapolator = Extrapolator
+        sd_test.extrapolator_kwargs = {
+            'method': 'constant',
+            'left': 0,
+            'right': 0
+        }
+
+    if sd_test.shape.interval > 5:
+        raise ValueError('Test spectral distribution interval is greater than'
+                         '5nm which is the maximum recommended value '
+                         'for computing the "CIE 2017 Colour Fidelity Index"!')
+
+    shape = SpectralShape(SPECTRAL_SHAPE_CIE2017.start,
+                          SPECTRAL_SHAPE_CIE2017.end, sd_test.shape.interval)
+
+    CCT, D_uv = CCT_reference_illuminant(sd_test)
+    sd_reference = sd_reference_illuminant(CCT, shape)
+
+    # NOTE: All computations except CCT calculation use the
+    # "CIE 1964 10 Degree Standard Observer".
+    cmfs_10 = MSDS_CMFS['CIE 1964 10 Degree Standard Observer'].copy().align(
+        shape)
+
+    sds_tcs = load_TCS_CIE2017(shape).align(shape)
+
+    test_tcs_colorimetry_data = tcs_colorimetry_data(sd_test, sds_tcs, cmfs_10)
+    reference_tcs_colorimetry_data = tcs_colorimetry_data(
+        sd_reference, sds_tcs, cmfs_10)
+
+    delta_E_s = np.empty(len(sds_tcs.labels))
+    for i in range(len(delta_E_s)):
```

Comment:
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=6023637)